cli/src/main/java/keywhiz/cli/commands/DeleteAction.java

```diff
@@ -116,4 +123,14 @@ public DeleteAction(DeleteActionConfig deleteActionConfig, KeywhizClient client)
         throw new IllegalArgumentException("Invalid delete type specified: " + type);
     }
   }
-}
\ No newline at end of file
+
+  private static void validateMode(String mode) {
+    if (mode != null && !SecretDeletionMode.values().contains(mode)) {
+      throw new IllegalArgumentException(
+          format(
+              "Invalid mode [%s], must be one of %s.",
+              mode,
+              SecretDeletionMode.values()));
+    }
+  }
```

Comment:
The method was starting to get a bit lengthy with multiple blocks, so I opted to do validation via a helper method instead in order to keep things concise.